gxa/src/main/java/uk/ac/ebi/atlas/controllers/rest/DASFeaturesControllerNew.java

```diff
@@ -0,0 +1,126 @@
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.SetMultimap;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.springframework.context.annotation.Scope;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.ModelAndView;
+import uk.ac.ebi.atlas.controllers.JsonExceptionHandlingController;
+import uk.ac.ebi.atlas.model.AssayGroup;
+import uk.ac.ebi.atlas.model.experiment.Experiment;
+import uk.ac.ebi.atlas.model.experiment.baseline.BioentityPropertyName;
+import uk.ac.ebi.atlas.model.experiment.baseline.Factor;
+import uk.ac.ebi.atlas.model.experiment.baseline.impl.FactorSet;
+import uk.ac.ebi.atlas.model.experiment.differential.Contrast;
+import uk.ac.ebi.atlas.model.experiment.differential.DifferentialExpression;
+import uk.ac.ebi.atlas.model.experiment.differential.microarray.MicroarrayExpression;
+import uk.ac.ebi.atlas.search.SemanticQuery;
+import uk.ac.ebi.atlas.search.SemanticQueryTerm;
+import uk.ac.ebi.atlas.search.analyticsindex.differential.DifferentialAnalyticsSearchDao;
+import uk.ac.ebi.atlas.search.analyticsindex.differential.DifferentialAnalyticsSearchService;
+import uk.ac.ebi.atlas.search.diffanalytics.DiffAnalytics;
+import uk.ac.ebi.atlas.search.diffanalytics.DiffAnalyticsQueryBuilder;
+import uk.ac.ebi.atlas.search.diffanalytics.DiffAnalyticsSearchService;
+import uk.ac.ebi.atlas.species.SpeciesFactory;
+import uk.ac.ebi.atlas.trader.ContrastTrader;
+import uk.ac.ebi.atlas.trader.ExperimentTrader;
+
+import javax.inject.Inject;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+@Controller
+@Scope("request")
+public class DASFeaturesControllerNew {
+
+
+    private ExperimentTrader experimentTrader;
+    private DifferentialAnalyticsSearchService differentialAnalyticsSearchService;
+    private DifferentialJsonResultsParser differentialJsonResultsParser;
+
+
+    @Inject
+    public DASFeaturesControllerNew(ExperimentTrader experimentTrader,
+                                    DifferentialAnalyticsSearchService differentialAnalyticsSearchService,
+                                    DifferentialJsonResultsParser differentialJsonResultsParser) {
+
+        this.experimentTrader = experimentTrader;
+        this.differentialAnalyticsSearchService = differentialAnalyticsSearchService;
+        this.differentialJsonResultsParser = differentialJsonResultsParser;
+    }
+
+    @ResponseStatus(value = HttpStatus.INTERNAL_SERVER_ERROR)
+    public ModelAndView InternalServerHandleException(Exception e) {
```

Comment:
I like that this exists. Consider, though, that the endpoint is in XML, and it's used programmatically - probably the client will parse it to get individual fields - maybe it's better to show an error message in the same kind of format?
Also maybe it's not the best to show external users our internal messages - how about logging the error message in our logs, and saying something like "Unexpected error occurred, email us if this persists"?